llvm/test/CodeGen/EraVM/split-loop-phi-live-ranges.ll

```diff
@@ -29,19 +29,19 @@ define i256 @test() {
 ; CHECK-NEXT:    [[ADD2:%.*]] = add nuw nsw i256 [[RESULT]], 64
 ; CHECK-NEXT:    [[INTTOPTR0:%.*]] = inttoptr i256 [[ADD2]] to ptr addrspace(1)
 ; CHECK-NEXT:    store i256 0, ptr addrspace(1) [[INTTOPTR0]], align 1
-; CHECK-NEXT:    [[ADD3:%.*]] = add nuw nsw i256 [[RESULT]], -64
+; CHECK-NEXT:    [[ADD3:%.*]] = add i256 [[RESULT]], -64
```

Comment:
Shouldn't we keep flags here?
Saving them below would take more effort but given `ADD3 = add nuw, nsw` and original `ADD4 = add nuw, nsw` there must be no wrap even when substituting `RESULT` with `ADD3`.

Reply:
I found cases with alive2, where not removing flags from the NearestDominator is leading to issues, like [here](https://alive2.llvm.org/ce/z/-KVMhN).